idx/memory/memory_find_test.go

```diff
@@ -223,3 +290,92 @@ func BenchmarkConcurrent8Find(b *testing.B) {
 	}
 	close(ch)
 }
+
+func ixFindByTag(org, q int) {
+	series, err := ix.IdsByTagExpressions(org, tagQueries[q].Expressions, 0)
+	if err != nil {
+		panic(err)
+	}
+	if len(series) != tagQueries[q].ExpectedResults {
+		for s := range series {
+			memoryIdx := ix.(*MemoryIdx)
+			fmt.Println(memoryIdx.DefById[s.ToString()].Tags)
+		}
+		panic(fmt.Sprintf("%+v expected %d got %d results instead", tagQueries[q].Expressions, tagQueries[q].ExpectedResults, len(series)))
+	}
+}
+
+func BenchmarkTagFindSimpleIntersect(b *testing.B) {
+	if ix == nil {
+		Init()
+	}
+	b.ReportAllocs()
+	b.ResetTimer()
+	for n := 0; n < b.N; n++ {
+		q := n % 2
+		org := (n % 2) + 1
+		ixFindByTag(org, q)
+	}
+}
+
+func BenchmarkTagFindRegexIntersect(b *testing.B) {
+	if ix == nil {
+		Init()
+	}
+	b.ReportAllocs()
+	b.ResetTimer()
+	for n := 0; n < b.N; n++ {
+		q := (n % 2) + 2
+		org := (n % 2) + 1
+		ixFindByTag(org, q)
+	}
+}
+
+func BenchmarkTagFindMatchingAndFiltering(b *testing.B) {
+	if ix == nil {
+		Init()
+	}
+	b.ReportAllocs()
+	b.ResetTimer()
+	for n := 0; n < b.N; n++ {
+		q := (n % 2) + 4
+		org := (n % 2) + 1
+		ixFindByTag(org, q)
+	}
+}
+
+func BenchmarkTagFindMatchingAndFilteringWithRegex(b *testing.B) {
+	if ix == nil {
+		Init()
+	}
+	b.ReportAllocs()
+	b.ResetTimer()
+	for n := 0; n < b.N; n++ {
+		q := (n % 2) + 6
+		org := (n % 2) + 1
+		ixFindByTag(org, q)
+	}
+}
+
+func BenchmarkTagQueryFilterAndIntersect(b *testing.B) {
+	if ix == nil {
+		Init()
+	}
+	q := tagQuery{Expressions: []string{"direction!=~read", "device!=", "host=~host9[0-9]0", "dc=dc1", "disk!=disk1", "metric=disk_time"}, ExpectedResults: 90}
+
+	b.ReportAllocs()
+	b.ResetTimer()
+
+	for n := 0; n < b.N; n++ {
+		series, err := ix.IdsByTagExpressions(1, q.Expressions, 150000)
+		if err != nil {
+			panic(err)
+		}
+		if len(series) != q.ExpectedResults {
+			for _, s := range series {
+				fmt.Println(s)
+			}
+			panic(fmt.Sprintf("%+v expected %d got %d results instead", q.Expressions, q.ExpectedResults, len(series)))
```

Comment:
don't print to stdout and call panic.
use the standard functions https://golang.org/pkg/testing/#TB

Reply:
fixed https://github.com/grafana/metrictank/pull/729/commits/186deabb44a6eb6966bfa7e1b95c147b3fc5f22a